exercises/queen-attack/src/lib.rs

```diff
@@ -1 +1,19 @@
+pub struct ChessPosition;
 
+pub struct Queen;
+
+impl ChessPosition {
+    pub fn new(rank: i32, file: i32) -> Result<Self, i32> {
```

Comment:
`Result<Self, i32>` is ambiguous: what should the return value be when someone calls `ChessPosition::new(-2, -1)`?

There are several ways in which we might handle this in the real world: use a builder pattern, so it's unambiguous  which step raised an error; return an enum with `Rank`, `File`, and `Both` variants, and more. Both of those possibilities require more work from the student than I believe this exercise should.

Instead, would you be willing to update the stub, example, and test suite such that `new` returns `Option<Self>`?